components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/resources/UserResource.java

```diff
@@ -338,8 +341,20 @@ public Response getAuthorizedUser(
             SCIMResponse scimResponse = null;
             scimResponse = userResourceEndpoint.listByFilter(filter, userManager, format);
 
-            return new JAXRSResponseBuilder().buildResponse(scimResponse);
+            // get scim id to retrieve user information
+            String scimId;
+            JSONObject responseObject = new JSONObject(scimResponse.getResponseMessage());
+            JSONArray resourceArray = responseObject.getJSONArray(SCIMConstants.ListedResourcesConstants.RESOURCES);
+            scimId = resourceArray.getJSONObject(0).getString(SCIMConstants.CommonSchemaConstants.ID);
+
+            SCIMResponse userDataResponse = userResourceEndpoint.get(scimId, format, userManager);
+            return new JAXRSResponseBuilder().buildResponse(userDataResponse);
 
+        } catch (JSONException e) {
+            logger.error("Error while processing the request", e);
```

Comment:
do we need to log the error here? I think handleCharonException create a debug log already..